src/pages/SingleVideo/SingleVideo.jsx

```diff
@@ -0,0 +1,154 @@
+import { useState, useEffect } from 'react';
+import ReactPlayer from 'react-player/youtube';
+import { useParams } from 'react-router-dom';
+import { PlaylistModal } from '../../components';
+import { useAuth, useLikes, useWatchLater } from '../../context';
+import { addToWatchLater, deleteLike, deleteWatchLater, getVideo, likesHandler } from '../../utils';
+import styles from './SingleVideo.module.css';
+
+const SingleVideo = () => {
+	const [currentVideo, setCurrentVideo] = useState(null);
+	const [isLoading, setIsLoading] = useState(true);
```

Comment:
Can isLoading and btnLoading be merged into one object?

Reply:
`isLoading` is for the page initial load and `btnLoading` is to avoid the user to double click any button and send unnecessary api calls. Thanks for the suggestion, I will try and merge it in a refactor later on 🚀